oc-game-swift/Players/Player.swift

```diff
@@ -36,35 +42,166 @@ class Player {
-                        player.heroTeam.append(Dwarf(name: name, type: .dwarf))
+                        player.heroTeam.append(Dwarf(name: name))
                     case .colossus:
                         // Add a Colossus
-                        player.heroTeam.append(Colossus(name: name, type: .colossus))
+                        player.heroTeam.append(Colossus(name: name))
                     }
                     // Print that all is good
                     print("Your \(choice) \(name) has been added to your team")
+                    // Add the name to the array of heroes names
+                    Game.heroesNames.append(name)
                     // Turn the check on true
                     heroWasCreated = true
-                    
-                } else {
-                    // If the name is empty
-                    print("Srsly? Everytime? Choose a name...")
-                    
                 }
             }
         }
     }
     
+    /// Choosing a hero
+    /// - Player have to choose a hero to fight with this round
+    /// - return: The selected hero
+    func chooseHeroFromYourTeam() -> Hero? {
+        // Hero is choose ?
+        var heroChosen = false
+        // While not
+        while heroChosen == false {
+            // Print the message
+            print("-----------------------------------"
+                + "\n\(self.name), it's your turn. Choose a hero to fight with"
+                + "\n-----------------------------------"
+                + "\n1. \(self.heroTeam[0].description())"
+                + "\n2. \(self.heroTeam[1].description())"
+                + "\n3. \(self.heroTeam[2].description())")
+            // Choose a hero, and check if its ok
+            if let hero = chooseHero(self) {
+                heroChosen = true
+                return hero
+            }
+        }
+    }
+    
+    /// Player have to choose a hero to attack
+    /// - player: Player, The one who chose
+    /// - return: Hero?, The selected hero
+    func chooseHeroToAttack(_ player: Player) -> Hero? {
+        // Hero is choose ?
+        var heroChosen = false
+        // While not
+        while heroChosen == false {
+            // Print the message
+            print("-----------------------------------"
+                + "\nChoose a target from the team of \(player.name)"
+                + "\n-----------------------------------"
+                + "\n1. \(player.heroTeam[0].description())"
+                + "\n2. \(player.heroTeam[1].description())"
+                + "\n3. \(player.heroTeam[2].description())")
+            
+            // Choose a hero
+            if let targetHero = chooseHero(player) {
+                heroChosen = true
+                return targetHero
+            }
+        }
+    }
+    
+    /// Player have to choose a hero to heal with the healer
+    /// - return: Hero?, The selected hero
+    func chooseHeroToHeal() -> Hero? {
```

Comment:
you should try to factorize with previous method since this seems to be the same code

Reply:
Done.